src/compas_model/elements/element.py

```diff
@@ -6,97 +6,134 @@
     if TYPE_CHECKING:
         from compas_model.model import ElementNode  # noqa: F401
 
-from abc import abstractmethod
-
 import compas.geometry
 import compas.datastructures  # noqa: F401
 
 from compas.data import Data
-from compas.geometry import Frame  # noqa: F401
+from compas.geometry import Transformation
+
+
+class Feature(Data):
+    """Base class for all element features.
+
+    Parameters
+    ----------
+    name : str, optional
+        The name of the feature.
+
+    """
+
+    @property
+    def __data__(self):
+        # type: () -> dict
+        return {}
+
+    def __init__(self, name=None):
+        # type: (str | None) -> None
+        super(Feature, self).__init__(name=name)
+
+    def apply(self, shape):
+        # type: (compas.datastructures.Mesh | compas.geometry.Brep) -> compas.datastructures.Mesh | compas.geometry.Brep
```

Comment:
Why is a shape either a mesh or a brep? It can also be other types that are not closed solids. Often this data is only minimally visualized and refers to a larger set of custom attributes.

Do features always pertain to fabrication, or can they also relate to the mechanical properties of solvers involved in structural analysis?